apps/04TowersOfHanoi.js

```diff
@@ -3,91 +3,170 @@
 var assert = require('assert');
 var readline = require('readline');
 var rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
+    input: process.stdin,
+    output: process.stdout
 });
 
+var num = 0;
 var stacks = {
-  a: [4, 3, 2, 1],
-  b: [],
-  c: []
+    a: [4,3,2,1],
+    b: [],
+    c: []
 };
 
 function printStacks() {
-  console.log("a: " + stacks.a);
-  console.log("b: " + stacks.b);
-  console.log("c: " + stacks.c);
+    console.log("a: " + stacks.a);
+    console.log("b: " + stacks.b);
+    console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+function movePiece(startStack, endStack) {
+    // Your code here
+
+    //check for legal move first.
+    var answer = isLegal(startStack, endStack);
+    //if return is NOT false then pop/push
+      if (answer != false) {
+        var token = stacks[startStack].pop();
+        stacks[endStack].push(token);
+        countMove();
+        checkForWin();
+
+        // if all other rules fail
+    } else {
+      countMove();
+        console.log("Sorry buster, not allowed! Rules are that startStack < EndStack & can't be empty ")
+        return false;
+    }
+}
 
+function countMove(){
+    num++;
+    console.log(num + ' Number of Moves thus far');
+    return true;
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack, sameVal) {
+    // Your code here
+
+    var value1 = stacks[startStack].length - 1; //assign value of the array startStack
+    var value2 = stacks[endStack].length - 1; //assign value of the array endStack
+
+// evaluate if startStack is empty
+    if (value1 === -1) {
```

Comment:
Good data scrubbing here! 

I would recommend consolidating them using the `&&` or `||` operator though.